File: templates/java/v1/app/Main.java
package app;

import decorator.Perf;
import decorator.PerfTracker;
import factory.Shape;
import factory.ShapeFactory;
import facade.FacadeResult;
import facade.GreetingFacade;
import flyweight.StyledText;
import flyweight.TextStyle;
import flyweight.TextStyleFactory;
import composite.ElementNode;
import composite.TextNode;
import bridge.EmailNotifier;
import bridge.JsonFormatter;
import bridge.SmsNotifier;
import bridge.TextFormatter;
import adapter.LegacyLogger;
import adapter.LegacyLoggerAdapter;
import adapter.Logger;
import builder.HttpRequest;
import prototype.Document;
import singleton.AppConfig;
import chain.AuthHandler;
import chain.BusinessHandler;
import chain.Request;
import chain.ValidationHandler;
import command.AppendTextCommand;
import command.CommandInvoker;
import command.TextBuffer;
import iterator.Item;
import iterator.ItemCollection;
import mediator.ChatRoomMediator;
import mediator.UserParticipant;
import memento.TextEditor;
import memento.TextSnapshot;
import observer.EventBus;
import observer.LoggingObserver;
import state.DocumentContext;
import state.DraftState;
import state.PublishedState;
import templatemethod.DetailedReportGenerator;
import templatemethod.SummaryReportGenerator;
import visitor.ElementNode;
import visitor.RenderVisitor;
import visitor.TextNode;
import factorymethod.EmailNotificationCreator;
import factorymethod.SmsNotificationCreator;
import abstractfactory.DarkUiFactory;
import abstractfactory.LightUiFactory;
import abstractfactory.UiFactory;
import proxy.GreeterProxyFactory;
import proxy.GreeterService;
import proxy.GreeterServiceImpl;
import strategy.FormalStrategy;
import strategy.FriendlyStrategy;
import strategy.GreeterContext;
import strategy.UppercaseStrategy;

import java.lang.reflect.Method;

public class Main {
    static class Greeter {
        @Perf(name = "Greeter.hello")
        public String hello(String name) {
            String message = "Hello world from " + name;
            return message;
        }
    }

    public static void main(String[] args) throws Exception {
        Greeter g = new Greeter();

        Method m = Greeter.class.getMethod("hello", String.class);
        String result = (String) PerfTracker.invoke(g, m, "{{projectName}}");

        System.out.println(result);

        Shape shape = ShapeFactory.create("circle");
        System.out.println(shape.draw());

        GreeterContext context = new GreeterContext(new FriendlyStrategy());
        System.out.println(context.greet("{{projectName}}"));

        context.setStrategy(new FormalStrategy());
        System.out.println(context.greet("{{projectName}}"));

        context.setStrategy(new UppercaseStrategy());
        System.out.println(context.greet("{{projectName}}"));

        GreetingFacade facade = new GreetingFacade();
        FacadeResult result = facade.greetAndDraw("{{projectName}}", new FormalStrategy(), "square");
        System.out.println(result.getGreeting());
        System.out.println(result.getDrawing());

        TextStyleFactory styleFactory = new TextStyleFactory();
        TextStyle headline = styleFactory.getStyle("Inter", 18, "#111111", true, false);
        TextStyle body = styleFactory.getStyle("Inter", 14, "#333333", false, false);
        TextStyle headlineShared = styleFactory.getStyle("Inter", 18, "#111111", true, false);

        StyledText line1 = new StyledText("Flyweight headline", headline);
        StyledText line2 = new StyledText("Flyweight body", body);
        StyledText line3 = new StyledText("Headline reused", headlineShared);

        System.out.println(line1.render());
        System.out.println(line2.render());
        System.out.println(line3.render());
        System.out.println("Flyweight cache size: " + styleFactory.cacheSize());

        GreeterService service = new GreeterServiceImpl();
        GreeterService timedService = GreeterProxyFactory.createTimed(service);
        System.out.println(timedService.greet("{{projectName}}"));

        ElementNode root = new ElementNode("div")
            .add(new TextNode("Composite root: "))
            .add(new ElementNode("span")
                .add(new TextNode("child")));
        System.out.println(root.render());

        EmailNotifier emailText = new EmailNotifier(new TextFormatter());
        EmailNotifier emailJson = new EmailNotifier(new JsonFormatter());
        SmsNotifier smsText = new SmsNotifier(new TextFormatter());

        System.out.println(emailText.notify("Bridge message"));
        System.out.println(emailJson.notify("Bridge message"));
        System.out.println(smsText.notify("Bridge message"));

        Logger logger = new LegacyLoggerAdapter(new LegacyLogger());
        logger.log("INFO", "Adapter message");

        HttpRequest request = new HttpRequest.Builder()
            .url("https://api.example.com/messages")
            .method("POST")
            .header("Content-Type", "application/json")
            .header("Accept", "application/json")
            .body("{\"message\":\"Hello\"}")
            .timeoutMs(3000)
            .build();
        System.out.println(request.summary());

        Document baseDoc = new Document("Prototype", java.util.List.of("Intro"));
        Document copy = baseDoc.clonePrototype();
        copy.setTitle("Prototype Copy");
        copy.addParagraph("Second paragraph");
        System.out.println(baseDoc.summary());
        System.out.println(copy.summary());

        AppConfig config = AppConfig.getInstance();
        config.set("env", "dev");
        System.out.println("Singleton env: " + config.get("env", "local"));

        ValidationHandler validation = new ValidationHandler();
        AuthHandler auth = new AuthHandler();
        BusinessHandler business = new BusinessHandler();
        validation.linkWith(auth).linkWith(business);

        Request request = new Request("req-1", "token:abc; payload data");
        System.out.println(validation.handle(request));

        TextBuffer buffer = new TextBuffer();
        CommandInvoker invoker = new CommandInvoker();
        System.out.println(invoker.run(new AppendTextCommand(buffer, "Hello")));
        System.out.println(invoker.run(new AppendTextCommand(buffer, " Command")));
        System.out.println("Command history size: " + invoker.historySize());

        ItemCollection collection = new ItemCollection();
        collection.add(new Item("1", "Alpha"));
        collection.add(new Item("2", "Beta"));
        for (Item item : collection) {
            System.out.println("Iterator item: " + item.getId() + "=" + item.getName());
        }

        ChatRoomMediator room = new ChatRoomMediator();
        UserParticipant alice = new UserParticipant("alice", room);
        UserParticipant bob = new UserParticipant("bob", room);
        room.register(alice);
        room.register(bob);
        alice.send("Mediator hello");

        TextEditor editor = new TextEditor();
        editor.append("Hello");
        TextSnapshot snapshot = editor.save();
        editor.append(" Memento");
        System.out.println("Memento current: " + editor.current());
        editor.restore(snapshot);
        System.out.println("Memento restored: " + editor.current());

        EventBus bus = new EventBus();
        LoggingObserver obsA = new LoggingObserver("A");
        LoggingObserver obsB = new LoggingObserver("B");
        bus.subscribe(obsA);
        bus.subscribe(obsB);
        bus.publish("message.sent", "Observer hello");

        DocumentContext doc = new DocumentContext(new DraftState());
        System.out.println("State " + doc.currentState() + ": " + doc.process("State message"));
        doc.setState(new PublishedState());
        System.out.println("State " + doc.currentState() + ": " + doc.process("State message"));

        SummaryReportGenerator summary = new SummaryReportGenerator();
        DetailedReportGenerator detailed = new DetailedReportGenerator();
        System.out.println(summary.generate("Template Method report"));
        System.out.println(detailed.generate("Line one\nLine two"));

        ElementNode root = new ElementNode("div")
            .add(new TextNode("Visitor root: "))
            .add(new ElementNode("span").add(new TextNode("child")));
        RenderVisitor renderer = new RenderVisitor();
        System.out.println(root.accept(renderer));

        EmailNotificationCreator emailCreator = new EmailNotificationCreator();
        SmsNotificationCreator smsCreator = new SmsNotificationCreator();
        System.out.println(emailCreator.notify("Factory Method message"));
        System.out.println(smsCreator.notify("Factory Method message"));

        UiFactory light = new LightUiFactory();
        UiFactory dark = new DarkUiFactory();
        System.out.println(light.createButton().render() + "," + light.createCheckbox().render());
        System.out.println(dark.createButton().render() + "," + dark.createCheckbox().render());
    }
}

File: templates/python/v1/app/main.py
from decorator.perf import profile
from factory.shape_factory import ShapeFactory
from facade.greeting_facade import GreetingFacade
from flyweight.text_style_factory import TextStyleFactory, StyledText
from composite.nodes import ElementNode, TextNode
from proxy.timed_proxy import TimedProxy
from bridge.formatters import JsonFormatter, TextFormatter
from bridge.notifiers import EmailNotifier, SmsNotifier
from adapter.legacy_logger import LegacyLogger
from adapter.logger_adapter import LegacyLoggerAdapter
from builder.http_request_builder import HttpRequestBuilder
from prototype.document_prototype import DocumentPrototype
from singleton.app_config import AppConfig
from chain.handlers import ValidationHandler, AuthHandler, BusinessHandler, Request
from command.commands import TextBuffer, AppendTextCommand, CommandInvoker
from iterator.collection import ItemCollection, Item
from mediator.chat_mediator import ChatMediator, UserParticipant
from memento.text_editor import TextEditor
from observer.event_bus import EventBus, LoggingObserver
from state.states import DocumentContext, DraftState, PublishedState
from templatemethod.report_generators import SummaryReportGenerator, DetailedReportGenerator
from visitor.nodes import ElementNode, TextNode, RenderVisitor
from factorymethod.notifications import EmailNotificationCreator, SmsNotificationCreator
from abstractfactory.ui_factory import LightUiFactory, DarkUiFactory
from strategy.greeter_strategies import (
    FriendlyStrategy,
    FormalStrategy,
    UppercaseStrategy,
    GreeterContext,
)


class Greeter:
    @profile()
    def hello(self, name: str) -> str:
        message = f"Hello world from {name}"
        return message


def main() -> None:
    g = Greeter()
    print(g.hello("{{projectName}}"))

    shape = ShapeFactory.create("circle")
    print(shape.draw())

    context = GreeterContext(FriendlyStrategy())
    print(context.greet("{{projectName}}"))

    context.set_strategy(FormalStrategy())
    print(context.greet("{{projectName}}"))

    context.set_strategy(UppercaseStrategy())
    print(context.greet("{{projectName}}"))

    facade = GreetingFacade()
    result = facade.greet_and_draw("{{projectName}}", FormalStrategy(), "square")
    print(result.greeting)
    print(result.drawing)

    style_factory = TextStyleFactory()
    headline = style_factory.get_style("Inter", 18, "#111111", True, False)
    body = style_factory.get_style("Inter", 14, "#333333", False, False)
    headline_shared = style_factory.get_style("Inter", 18, "#111111", True, False)

    line1 = StyledText("Flyweight headline", headline)
    line2 = StyledText("Flyweight body", body)
    line3 = StyledText("Headline reused", headline_shared)

    print(line1.render())
    print(line2.render())
    print(line3.render())
    print(f"Flyweight cache size: {style_factory.cache_size()}")

    class GreeterService:
        def greet(self, name: str) -> str:
            return f"Hello {name} from proxy"

    greeter = GreeterService()
    timed_greeter = TimedProxy(greeter)
    print(timed_greeter.greet("{{projectName}}"))

    root = ElementNode("div")
    root.add(TextNode("Composite root: "))
    root.add(ElementNode("span").add(TextNode("child")))
    print(root.render())

    email_text = EmailNotifier(TextFormatter())
    email_json = EmailNotifier(JsonFormatter())
    sms_text = SmsNotifier(TextFormatter())

    print(email_text.notify("Bridge message"))
    print(email_json.notify("Bridge message"))
    print(sms_text.notify("Bridge message"))

    logger = LegacyLoggerAdapter(LegacyLogger())
    logger.log("INFO", "Adapter message")

    request = (
        HttpRequestBuilder()
        .set_url("https://api.example.com/messages")
        .set_method("POST")
        .add_header("Content-Type", "application/json")
        .add_header("Accept", "application/json")
        .set_body("{\"message\":\"Hello\"}")
        .set_timeout_ms(3000)
        .build()
    )
    print(request.summary())

    base_doc = DocumentPrototype("Prototype", ["Intro"])
    copy = base_doc.clone_prototype()
    copy.title = "Prototype Copy"
    copy.add_paragraph("Second paragraph")
    print(base_doc.summary())
    print(copy.summary())

    config = AppConfig()
    config.set("env", "dev")
    print(f"Singleton env: {config.get('env', 'local')}")

    validation = ValidationHandler()
    auth = AuthHandler()
    business = BusinessHandler()
    validation.link_with(auth).link_with(business)

    request = Request("req-1", "token:abc; payload data")
    print(validation.handle(request))

    buffer = TextBuffer()
    invoker = CommandInvoker()
    print(invoker.run(AppendTextCommand(buffer, "Hello")))
    print(invoker.run(AppendTextCommand(buffer, " Command")))
    print(f"Command history size: {invoker.history_size()}")

    collection = ItemCollection()
    collection.add(Item("1", "Alpha"))
    collection.add(Item("2", "Beta"))
    for item in collection:
        print(f"Iterator item: {item.id}={item.name}")

    room = ChatMediator()
    alice = UserParticipant("alice", room)
    bob = UserParticipant("bob", room)
    room.register(alice)
    room.register(bob)
    alice.send("Mediator hello")

    editor = TextEditor()
    editor.append("Hello")
    snapshot = editor.save()
    editor.append(" Memento")
    print(f"Memento current: {editor.current()}")
    editor.restore(snapshot)
    print(f"Memento restored: {editor.current()}")

    bus = EventBus()
    obs_a = LoggingObserver("A")
    obs_b = LoggingObserver("B")
    bus.subscribe(obs_a)
    bus.subscribe(obs_b)
    bus.publish("message.sent", "Observer hello")

    doc = DocumentContext(DraftState())
    print(f"State {doc.current_state()}: {doc.process('State message')}")
    doc.set_state(PublishedState())
    print(f"State {doc.current_state()}: {doc.process('State message')}")

    summary = SummaryReportGenerator()
    detailed = DetailedReportGenerator()
    print(summary.generate("Template Method report"))
    print(detailed.generate("Line one\nLine two"))

    root = ElementNode("div")
    root.add(TextNode("Visitor root: "))
    root.add(ElementNode("span").add(TextNode("child")))
    renderer = RenderVisitor()
    print(root.accept(renderer))

    email_creator = EmailNotificationCreator()
    sms_creator = SmsNotificationCreator()
    print(email_creator.notify("Factory Method message"))
    print(sms_creator.notify("Factory Method message"))

    light = LightUiFactory()
    dark = DarkUiFactory()
    print(f"{light.create_button().render()},{light.create_checkbox().render()}")
    print(f"{dark.create_button().render()},{dark.create_checkbox().render()}")


if __name__ == "__main__":
    main()